### PR TITLE
Fix provisioning and key import policy on OPTIGA.

### PR DESCRIPTION
**Pull request by dcgaws**
Rationalize key import policy settings. Fix public key slot location. Improve key provisioning flow.

### CODE REVIEW

File: demos/dev_mode_key_provisioning/README.md [hunk 1]
```diff
@@ -36,6 +45,12 @@ Use openssl to format the binary encoded (DER) device public key as PEM:
 openssl ec -inform der -in DevicePublicKeyDer.bin -pubin -pubout -outform pem -out DevicePublicKey.pem
 ```
 
+Don't forget to disable the temporary key generation setting you enabled above. Otherwise, the device will create yet another keyset, and you will have to repeat the previous steps:
```
Comment (alexa-noxon) [inline demos/dev_mode_key_provisioning/README.md]: Is the use of the word keyset vs keypair meaningful?

File: demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c [hunk 2]
```diff
@@ -1027,24 +1031,34 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
...
+        }
 
-                prvExportPublicKey( xSession,
-                                    xProvisionedState.xPublicKey,
-                                    &xProvisionedState.pucDerPublicKey,
-                                    &xProvisionedState.ulDerPublicKeyLength );
-            }
+        /* Ignore errors. If any of the above objects couldn't be read, try to
+         * generate new ones below. */
+        xResult = CKR_OK;
```
Comment (alexa-noxon) [inline demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c]: Why are we ignoring this/do we use the return of prvGetProvsionedState for anything?  

Even if object are not found, the result would still be CKR_OK, right?  What is the error that is returned that we want to ignore?

File: demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c [hunk 3]
```diff
@@ -1027,24 +1031,34 @@ CK_RV xProvisionDevice( CK_SESSION_HANDLE xSession,
...
+        xResult = xProvisionGenerateKeyPairEC( xSession,
+                                               ( uint8_t * ) pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS,
+                                               ( uint8_t * ) pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS,
+                                               &xProvisionedState.xPrivateKey,
+                                               &xProvisionedState.xPublicKey );
+
+        /* Get the bytes of the new public key. */
+        if( CKR_OK == xResult )
+        {
+            prvExportPublicKey( xSession,
```
Comment (alexa-noxon) [inline demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c]: If we have a keypair already generated, and then we also use the FORCE GENERATE option, is there any chance we double allocate the public key bytes here (first in prvGetProvisionedState above, then again here)?

File: demos/dev_mode_key_provisioning/README.md (hunk 1 above)
Comment (dcgaws) [reply demos/dev_mode_key_provisioning/README.md]: Fixed, thank you.

File: demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c (hunk 2 above)
Comment (dcgaws) [reply demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c]: I updated the inline comment.

File: demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c (hunk 3 above)
Comment (dcgaws) [reply demos/dev_mode_key_provisioning/src/aws_dev_mode_key_provisioning.c]: Fixed, thank you.

**Pull request merged**